nextcord/application_command.py

```diff
@@ -0,0 +1,883 @@
+        own.
+
+        Parameters
+        ----------
+        state: :class:`ConnectionState`
+            State object used to fetch Guild, Channel, etc from cache.
+        interaction: :class:`Interaction`
+            Interaction object to pass to callback.
+        option_data: :class:`list`
+            List of option data dictionaries from interaction data payload.
+        """
+        if self.children:
+            # Discord currently does not allow commands that have subcommands to be ran. Therefore, if a command has
+            # children, a subcommand must be being called.
+            await self.children[option_data[0]["name"]].call(state, interaction, option_data[0].get("options", {}))
+        elif self.type in (ApplicationCommandType.chat_input, ApplicationCommandOptionType.sub_command):
+            # Slash commands are able to have subcommands, therefore that is handled here.
+            await self.call_invoke_slash(state, interaction, option_data)
+        else:
+            # Anything that can't be handled in here should be raised for ApplicationCommand to handle.
+            # TODO: Figure out how to hide this in exception trace log, devs don't need to see it.
+            raise InvalidCommandType(f"{self.type} is not a handled Application Command type.")
+
+    async def call_invoke_slash(self, state: ConnectionState, interaction: Interaction,
+                                option_data: List[Dict[str, Any]]):
+        """|coro|
+
+        This invokes the slash command implementation with the given raw option data to turn into proper kwargs for
+        the callback.
+
+        Parameters
+        ----------
+        state: :class:`ConnectionState`
+            State object used to fetch Guild, Channel, etc from cache.
+        interaction: :class:`Interaction`
+            Interaction object to pass to the callback.
+        option_data: :class:`list`
+            List of option data dictionaries from interaction data payload.
+        """
+        kwargs = {}
+        uncalled_args = self.arguments.copy()
+        for arg_data in option_data:
+            if arg_data["name"] in uncalled_args:
+                uncalled_args.pop(arg_data["name"])
+                kwargs[self.arguments[arg_data["name"]].functional_name] = \
+                    self.arguments[arg_data["name"]].handle_slash_argument(state, arg_data["value"], interaction)
+            else:
+                # TODO: Handle this better.
+                raise NotImplementedError(f"An argument was provided that wasn't already in the function, did you"
+                                          f"recently change it?\nRegistered Args: {self.arguments}, Discord-sent"
+                                          f"args: {interaction.data['options']}, broke on {arg_data}")
+        for uncalled_arg in uncalled_args.values():
+            kwargs[uncalled_arg.functional_name] = uncalled_arg.default
```

Comment:
I'm a tad out of it atm, but surely theres a way to remove this second ``for`` loop and do it in one. All for reduce ``O()``